airbyte-cdk/bulk/toolkits/load-avro/src/main/kotlin/io/airbyte/cdk/load/data/avro/AvroMapperPipelineFactory.kt

```diff
@@ -16,6 +16,15 @@ import io.airbyte.cdk.load.data.NullOutOfRangeIntegers
 import io.airbyte.cdk.load.data.SchemalessValuesToJsonString
 import io.airbyte.cdk.load.data.TimeStringToInteger
 
+/**
+ * @deprecated This class is deprecated and will be removed in a future release. We are
+ * transitioning to a new mapping pattern using EnrichedDestinationRecordAirbyteValue and
+ * EnrichedAirbyteValue for improved type safety and consistency.
+ *
+ * For examples on implementing the new pattern, please refer to:
+ * - DestinationRecordRaw.asEnrichedDestinationRecordAirbyteValue()
+ * - toRecord() method in the s3-data-lake destination
+ */
```

Comment:
for avro+parquet, we should also mention that their behavior doing recursion / deeply-typed structs is sad and we want to stop doing it